### DISCUSSION TITLE
How to notify provider that consumer is offline in callback parameter

### DISCUSSION THREAD

**Discussion by hehuaiping** [apache/dubbo#13872] [Question] [answered]
参考https://cn.dubbo.apache.org/zh-cn/overview/mannual/java-sdk/advanced-features-and-usage/service/callback-parameter/
1、客户端正常向服务端注册了listener，一段时间后客户端异常断电导致不能执行优雅停机，服务端被注册的listener如何才能被清理？
2、服务端异常断电，导致不能正常执行优雅停机。客户端如何感知到，并再重新注册listener

**Comment by AlbumenJ** [comment, marked as answer]
For this scenario, we recommend letting the consumer process publish a service for the provider to subscribe to.
